### ISSUE TITLE
TypeError: list indices must be integers or slices, not str

### ISSUE THREAD

**Issue by wernerhp** [cyberjunky/home-assistant-garmin_connect#200]
```
Logger: custom_components.garmin_connect
Source: helpers/update_coordinator.py:382
integration: Garmin Connect (documentation, issues)
First occurred: 21:12:50 (20 occurrences)
Last logged: 22:47:52

Unexpected error fetching garmin_connect data
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 382, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/garmin_connect/__init__.py", line 117, in _async_update_data
    summary = await self.hass.async_add_executor_job(
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/garminconnect/__init__.py", line 265, in get_user_summary
    if response["privacyProtected"] is True:
       ~~~~~~~~^^^^^^^^^^^^^^^^^^^^
TypeError: list indices must be integers or slices, not str
```
```
This error originated from a custom integration.

Logger: custom_components.garmin_connect
Source: helpers/update_coordinator.py:382
integration: Garmin Connect (documentation, issues)
First occurred: 21:07:50 (1 occurrences)
Last logged: 21:07:50

Unexpected error fetching garmin_connect data
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 382, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/garmin_connect/__init__.py", line 177, in _async_update_data
    sleep_score = sleep_data["dailySleepDTO"]["sleepScores"]["overall"]["value"]
                  ~~~~~~~~~~^^^^^^^^^^^^^^^^^
TypeError: list indices must be integers or slices, not str
```

**Comment by dgolovach** [comment]
Something seems to have changed on Garmin end, i'm also getting

> 2024-11-26 12:01:27.625 ERROR (MainThread) [custom_components.garmin_connect.config_flow] Unexpected exception
> Traceback (most recent call last):
>   File "/config/custom_components/garmin_connect/config_flow.py", line 47, in async_step_user
>     await self.hass.async_add_executor_job(api.login)
>   File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
>     result = self.fn(*self.args, **self.kwargs)
>              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.12/site-packages/garminconnect/__init__.py", line 181, in login
>     self.display_name = self.garth.profile["displayName"]
>                         ^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.12/site-packages/garth/http.py", line 104, in profile
>     assert isinstance(self._profile, dict)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> AssertionError

**Comment by jouster1974** [comment]
seeing a similar thing  here with 3 errors this evening

![image](https://github.com/user-attachments/assets/1c6b568f-6b88-4cfe-981f-ba1e4fc3b082)



This error originated from a custom integration.

Logger: custom_components.garmin_connect
Source: custom_components/garmin_connect/__init__.py:82
integration: Garmin Connect (documentation, issues)
First occurred: 23:27:55 (1 occurrences)
Last logged: 23:27:55

Unknown error occurred during Garmin Connect login request
Traceback (most recent call last):
  File "/config/custom_components/garmin_connect/__init__.py", line 82, in async_login
    await self.hass.async_add_executor_job(self._api.login)
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/garminconnect/__init__.py", line 181, in login
    self.display_name = self.garth.profile["displayName"]
                        ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/garth/http.py", line 104, in profile
    assert isinstance(self._profile, dict)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError

This error originated from a custom integration.

Logger: custom_components.garmin_connect
Source: helpers/update_coordinator.py:382
integration: Garmin Connect (documentation, issues)
First occurred: 19:11:01 (52 occurrences)
Last logged: 23:26:01

Unexpected error fetching garmin_connect data
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 382, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/garmin_connect/__init__.py", line 117, in _async_update_data
    summary = await self.hass.async_add_executor_job(
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/garminconnect/__init__.py", line 216, in get_user_summary
    if response["privacyProtected"] is True:
       ~~~~~~~~^^^^^^^^^^^^^^^^^^^^
TypeError: list indices must be integers or slices, not str


This error originated from a custom integration.

Logger: custom_components.garmin_connect
Source: helpers/update_coordinator.py:382
integration: Garmin Connect (documentation, issues)
First occurred: 19:05:59 (1 occurrences)
Last logged: 19:05:59

Unexpected error fetching garmin_connect data
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 382, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/garmin_connect/__init__.py", line 133, in _async_update_data
    alarms = await self.hass.async_add_executor_job(self._api.get_device_alarms)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/garminconnect/__init__.py", line 736, in get_device_alarms
    device_alarms = device_settings["alarms"]
                    ~~~~~~~~~~~~~~~^^^^^^^^^^
TypeError: list indices must be integers or slices, not str

**Comment by Bvdmgg** [comment]
Same problem here!

**Comment by perabo65** [comment]
Just as information, for a test I did try the hotfix/workaround mentioned in [https://github.com/matin/garth/issues/73](https://github.com/matin/garth/issues/73l) in the docker environment of Home Assistant:

1. Login to docker with docker exec -it <ID> /bin/bash
2. Change in /usr/local/lib/python3.12/site-packages/garth the file http.py as mentioned in the hotfix
3. Restart Home Assistant
4. Garmin integration is up and running again

Of course it's only a workaround and not a final solution, but for the moment I'm happy to have my body data back in HA

**Comment by cyberjunky** [comment]
Temp fixed in version 0.2.20!
